marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -24,60 +24,57 @@ const entityLib = require("/data-hub/5/impl/entity-lib.sjs");
  * trace logging being used. This ensures that the condition appears in logging, but it should not throw an error
  * because other entities in the search results may be able to have properties added for them.
  *
- * @param entityName
+ * @param entityNames
  * @param searchResponse
  */
-function addPropertiesToSearchResponse(entityName, searchResponse, propertiesToDisplay) {
+function addPropertiesToSearchResponse(entityNames, searchResponse, propertiesToDisplay) {
   const maxDefaultProperties = 5;
-  const selectedPropertyNames = typeof propertiesToDisplay === 'string' ? propertiesToDisplay.split(",") :  propertiesToDisplay;
-  const entityModel = entityLib.findModelByEntityName(entityName);
-  if (!entityModel) {
-    ds.throwServerError(`Could not add entity properties to search response; could not find an entity model for entity name: ${entityName}`);
-  }
-
-  const allMetadata = buildAllMetadata("", entityModel, entityName);
-  const propertyMetadata = allMetadata["allPropertiesMetadata"];
-
+  const selectedPropertyNames = typeof propertiesToDisplay === 'string' ? propertiesToDisplay.split(",") : propertiesToDisplay;
   let selectedPropertyMetadata = [];
-  if(selectedPropertyNames) {
-    selectedPropertyMetadata = buildSelectedPropertiesMetadata(allMetadata, selectedPropertyNames);
-  }
-  selectedPropertyMetadata = selectedPropertyMetadata.length > 0 ? selectedPropertyMetadata : propertyMetadata.slice(0, maxDefaultProperties);
+  let propertyMetadata = [];
+
+  if (Array.isArray(entityNames) && entityNames.length > 1) {
+    //'All Entities' option is selected
+    searchResponse.results.forEach(result => {
+      let entityInfo = {
```

Comment:
This initialization of data in "result" should happen in the function that's called. I think the forEach loop should be very simple and just do this:

```
searchResponse.results.forEach(result => {
  if (entityName != null) {
    addEntitySpecificProperties(...);
  } else {
    addGenericEntityProperties(...);
  }
});
```

That way, only addEntitySpecificProperties cares about selected properties - that concept doesn't exist in addGenericEntityProperties, since we're using a fixed set of columns (I think?).